data_validation/validation_builder.py

```diff
@@ -337,15 +338,21 @@ def get_source_query(self):
             source_input_query = self.get_query_from_file(
                 self.config_manager.source_query_file[0]
             )
-            source_aggregate_query = "SELECT "
-            for aggregate in self.config_manager.aggregates:
-                source_aggregate_query += self.get_aggregation_query(
-                    aggregate.get("type"), aggregate.get("target_column")
+            if "hash__all" in self._metadata:
+                calculated_query = CustomQueryRowBuilder().compile_custom_query(
+                    source_input_query, source_config
                 )
-            source_aggregate_query = self.get_wrapper_aggregation_query(
-                source_aggregate_query, source_input_query
-            )
-            query = self.source_client.sql(source_aggregate_query)
+                query = self.source_client.sql(calculated_query)
+            else:
+                source_aggregate_query = "SELECT "
+                for aggregate in self.config_manager.aggregates:
+                    source_aggregate_query += self.get_aggregation_query(
```

Comment:
Ideally `get_aggregation_query` should also be in the CustomQueryBuilder() class. There can be a CustomQueryBuilder().compile_row_query() and another for CustomQueryBuilder().compile_column_query().

This way, all custom query logic can be in the Custom query builder

Reply:
Done.